File: examples/migration-agent/python/classifier.py
from __future__ import annotations

SKIP_AUDIT_DIRS = frozenset({"node_modules", "dist"})


def classify_port_status(
    *,
    python_exists: bool,
    latest_ts_mtime: float,
    python_mtime: float,
    write_stubs: bool,
) -> str:
    if not python_exists:
        return "created" if write_stubs else "missing"

    return "stale" if latest_ts_mtime > python_mtime else "ok"


def should_skip_audit_dir(name: str) -> bool:
    return name in SKIP_AUDIT_DIRS

File: examples/migration-agent/ts/src/classifier.ts
export type PortAuditStatus = "ok" | "stale" | "missing" | "created";

export type ClassifyPortInput = {
  pythonExists: boolean;
  latestTsMtime: number;
  pythonMtime: number;
  writeStubs: boolean;
};

export function classifyPortStatus(input: ClassifyPortInput): PortAuditStatus {
  const { pythonExists, latestTsMtime, pythonMtime, writeStubs } = input;

  if (!pythonExists) {
    return writeStubs ? "created" : "missing";
  }

  return latestTsMtime > pythonMtime ? "stale" : "ok";
}

export const SKIP_AUDIT_DIRS = new Set(["node_modules", "dist"]);

export function shouldSkipAuditDir(name: string): boolean {
  return SKIP_AUDIT_DIRS.has(name);
}
